planning/autoware_obstacle_cruise_planner/src/node.cpp

```diff
@@ -1182,7 +1388,10 @@ std::optional<StopObstacle> ObstacleCruisePlannerNode::createStopObstacle(
   const auto & object_id = obstacle.uuid.substr(0, 4);
 
   // NOTE: consider all target obstacles when driving backward
-  if (!isStopObstacle(obstacle.classification.label)) {
+  if (
+    (obstacle.source_type == Obstacle::SourceType::PREDICTED_OBJECT &&
+     !isStopObstacle(obstacle.classification.label)) ||
+    (obstacle.source_type == Obstacle::SourceType::POINTCLOUD && !use_pointcloud_for_stop_)) {
```

Comment:
[imo] it's a little bit difficult to understand this condition.

This is my suggestion.

```c++
  if (
    (obstacle.source_type == Obstacle::SourceType::PREDICTED_OBJECT &&
     !isStopObstacle(obstacle.classification.label)) {
    return std::nullopt;
  }

  if (
    (obstacle.source_type == Obstacle::SourceType::POINTCLOUD && !use_pointcloud_for_stop_)) {
    return std::nullopt;
  }
```